packages/eas-cli/src/project/publish.ts

```diff
@@ -0,0 +1,308 @@
+    weba: 'audio/webm',
+    webm: 'video/webm',
+    webp: 'image/webp',
+    woff: 'font/woff',
+    woff2: 'font/woff2',
+    xhtml: 'application/xhtml+xml',
+    xls: 'application/vnd.ms-excel',
+    xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    xml: 'application/xml',
+    xul: 'application/vnd.mozilla.xul+xml',
+    zip: 'application/zip',
+    '7z': 'application/x-7z-compressed',
+  };
+
+  if (ext && ext in CONTENT_TYPE) {
+    return CONTENT_TYPE[ext];
+  }
+  return 'application/octet-stream'; // unrecognized extension
+}
+
+export function getBase64URLEncoding(buffer: Buffer): string {
+  const base64 = buffer.toString('base64');
+  return base64.replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '');
+}
+
+/**
+ * The storage key is used to store the asset in GCS
+ */
+export function getStorageKey(contentType: string, contentHash: string): string {
+  const nullSeparator = Buffer.alloc(1);
+  const hash = crypto
+    .createHash('sha256')
+    .update(contentType)
+    .update(nullSeparator)
+    .update(contentHash)
+    .digest();
+  return getBase64URLEncoding(hash);
+}
+
+/**
+ * Convenience function that computes an assets storage key starting from its buffer.
+ */
+export function getStorageKeyForAsset(asset: RawAsset): string {
+  const fileHash = getBase64URLEncoding(
+    crypto.createHash('sha256').update(asset['buffer']).digest()
+  );
+  const contentType = asset['contentType'];
+  return getStorageKey(contentType, fileHash);
+}
+
+export function convertAssetToUpdateInfoGroupFormat(asset: RawAsset): PartialManifestAsset {
+  const fileHash = getBase64URLEncoding(
+    crypto.createHash('sha256').update(asset['buffer']).digest()
+  );
+  const contentType = asset['contentType'];
+  const storageKey = getStorageKey(contentType, fileHash);
+  const bundleKey = [
+    crypto.createHash('md5').update(asset['buffer']).digest('hex'),
+    asset.type,
+  ].join('.');
+
+  return {
+    fileHash,
+    contentType,
+    storageBucket: STORAGE_BUCKET,
+    storageKey,
+    bundleKey,
+  };
+}
+
+export function buildUpdateInfoGroup(assets: CollectedAssets) {
+  let platform: PublishPlatforms;
+  const updateInfoGroup: any = {};
+  for (platform in assets) {
+    updateInfoGroup[platform] = {
+      launchAsset: convertAssetToUpdateInfoGroupFormat(assets[platform]?.launchAsset!),
+      assets: assets[platform]?.assets!.map(convertAssetToUpdateInfoGroupFormat),
+    };
+  }
+  return updateInfoGroup;
+}
+
+export function getDistRoot(customDist: string): string {
+  const projectRoot = process.cwd();
+  const distRoot = path.join(projectRoot, customDist);
+  if (!fs.existsSync(distRoot)) {
+    throw new Error(`${distRoot} does not exist. Please create it with your desired bundler.`);
+  }
+  return distRoot;
+}
+
+export function collectUserDefinedAssets(distRoot: string): RawAsset[] {
+  const assetRoot = path.join(distRoot, 'assets');
+  const assetPointers = Platforms.map(platform => {
```

Comment:
This is a good question and one that is still being discussed. @jonsamp has argued that a less error prone flow would be to make use of the new `branch` feature and make an `ios` branch that we publish to. This keeps the branches well typed and makes the reasoning a bit simpler. 

If we decide to add the parametrization, it shouldn't be too difficult, but at the very least should be done in a separate PR as it would require some server side changes.